app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostFragment.kt

```diff
@@ -268,15 +273,55 @@ class ZimHostFragment : BaseFragment(), ZimHostCallbacks, ZimHostContract.View {
   }
 
   private fun layoutServerStarted() {
-    serverTextView.text = getString(R.string.server_started_message, ip)
+    serverTextView.apply {
+      text = colorSocketAddress()
+      movementMethod = LinkMovementMethod.getInstance()
+    }
+    configureUrlSharingIcon()
     startServerButton.text = getString(R.string.stop_server_label)
     startServerButton.setBackgroundColor(resources.getColor(R.color.stopServerRed))
     bookDelegate.selectionMode = SelectionMode.NORMAL
     booksAdapter.notifyDataSetChanged()
   }
 
+  private fun colorSocketAddress(): SpannableString {
+    val spannableText = SpannableString(getString(R.string.server_started_message, ip))
+    val clickableText: ClickableSpan = object : ClickableSpan() {
+      override fun onClick(widget: View) {
+        startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(ip)))
```

Comment:
Is it better to add `android:autoLink="web"` in our xml file instead of this?

Reply:
I _don't_ think so :P 
Because we don't want to hyperlink the _entire_ textView, and the string we set to it is roughly of format `"some stuff" + ip` and if we add this xml attribute, then it'll hyperlink the whole textview, which isn't desired I believe.
Thoughts?

Reply:
I believe it only detects the link. [see](https://developer.android.com/reference/android/widget/TextView#attr_android:autoLink)

Reply:
Gotcha, thanks!